### PR TITLE
add leak canary

### PR DESCRIPTION
**Pull request by drEfrati**
* Main purpose:
 to add a tool which can detect leaks

* Technical description:

* Dilemmas you faced with?

* Tests

* Documentation (Source/readme.md)

### CODE REVIEW

File: sample-app/build.gradle [hunk 1]
```diff
@@ -70,6 +70,12 @@ dependencies {
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'com.android.support.test:runner:1.0.1'
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
+
+    debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.6.2'
+    releaseImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.6.2'
```
Comment (yosriz) [inline sample-app/build.gradle]: I think there's no need for no-op for release as this is a sample app, we would want to detect leaks event in release builds

**Comment by yosriz** [comment]
@drEfrati shouldn't we wait after fixing the leak? now it will bug us all way long while debugging

**Comment by drEfrati** [comment]
@yosriz 
Regrading the leak canary, yes we can wait for the leak fixes so if you approve this pull request then don't merge it yet.

File: sample-app/build.gradle (hunk 1 above)
Comment (yosriz) [reply sample-app/build.gradle]: you need to change it to `implementation` without specifying release/debug - probably CI is failing now as release cannot be built.

Comment (drEfrati) [reply sample-app/build.gradle]: fixed now

**Pull request merged**